### PR TITLE
feat(service): Add matrix service

### PR DESCRIPTION
**Pull request by arnocornette**
## Description

Adds a notifier service for [Matrix](https://matrix.org)
## Motivation and Context

This addresses #283 partially. I decided to split up the Matrix and Signal PRs as tey are two features in one ticket.

Matrix is an open standard and communication protocl for real-time communication.

## How Has This Been Tested?

I tested this in 2 steps:
 - Basic unit tests using mocks
 - Manual testing has been performed as well
## Screenshots / Output (if appropriate):
image
This is output from several runs.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (no code change)
- [ ] Refactor (refactoring production code)
- [ ] Other 

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by arnocornette** [comment]
Normally I reformated with gofumpt. My bad, didn't see that you used a different formatter

P.S. Maybe you can put it in the how to contribute page?

File: service/matrix/matrix.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package matrix
```
Comment (svaloumas) [inline service/matrix/matrix.go:1]: Hi @arnocornette !

Thanks a lot for your efforts and contribution! I had only some really minor comments, mainly stylish. Your implementation looks solid and in compliance with the repo's contribution guidelines, special thanks for that! 🙏 

To make this complete, could we add some documentation for the service as well? You can take a look at these examples. 
[example service README](https://github.com/nikoksr/notify/tree/main/service/whatsapp) & [example go doc](https://github.com/nikoksr/notify/blob/main/service/whatsapp/doc.go)

Comment (svaloumas) [reply service/matrix/matrix.go:1]: Your quick response to those comments is much-appreciated @arnocornette ! It seems that this [commit](https://github.com/nikoksr/notify/pull/410/commits/57c959c11c7596d4496ef9d4ae2bf73f36f4f014) somehow reverted the service [name](https://github.com/nikoksr/notify/pull/410/commits/57c959c11c7596d4496ef9d4ae2bf73f36f4f014#diff-73ab59a205b5c2707ca30efcfa6c13c94dfd8106c2bc2f77d0f39ffc8a14792eR2) :smile: 
Let's fix that together with [this](https://github.com/nikoksr/notify/pull/410/commits/57c959c11c7596d4496ef9d4ae2bf73f36f4f014#diff-73ab59a205b5c2707ca30efcfa6c13c94dfd8106c2bc2f77d0f39ffc8a14792eR2) link to point to the new service instead and I think we're perfect to go! :+1:

Comment (arnocornette) [reply service/matrix/matrix.go:1]: Hey @svaloumas not a problem! I think I fixed your last few remarks 😄 . Let me know if you find some more!

**Comment by svaloumas** [comment]
> Normally I reformatted with gofumpt. My bad, didn't see that you used a different formatter
> 
> P.S. Maybe you can put it in the how to contribute page?

@nikoksr Leaving this comment here cause I think that's a good idea to consider.

**Comment by nikoksr** [comment]
> > Normally I reformatted with gofumpt. My bad, didn't see that you used a different formatter
> > 
> > P.S. Maybe you can put it in the how to contribute page?
> 
> @nikoksr Leaving this comment here cause I think that's a good idea to consider.

100% agree. We can go over the step-by-step guide and make it a little bit more granular probably. Thanks for the feedback @arnocornette.

**Comment by arnocornette** [comment]
I think the linter will not complain now but it's hard as the lint action does not give the filenames it doesn't work on

**Comment by nikoksr** [comment]
@arnocornette not sure if you're aware of it, but you can run `make lint` and `make fmt` locally. That should help you resolve it.

**Review by svaloumas — APPROVED** [review]

**Comment by nikoksr** [comment]
Greatly appreciate your contribution, @arnocornette! And thanks again for the feedback.

**Comment by blackandred** [comment]
Awesome work, thank you! ❤️

**Pull request merged**